hkdf/src/hkdf.rs

```diff
@@ -263,11 +269,25 @@ mod tests {
             let ikm = hex::decode(&t.ikm).unwrap();
             let salt = hex::decode(&t.salt).unwrap();
             let info = hex::decode(&t.info).unwrap();
-            let mut hkdf = Hkdf::<Sha1>::extract(&salt[..], &ikm[..]);
+            let mut hkdf = Hkdf::<Sha1>::extract(Option::from(&salt[..]), &ikm[..]);
             let okm = hkdf.expand(&info[..], t.length);
 
             assert_eq!(hex::encode(hkdf.prk), t.prk);
             assert_eq!(hex::encode(okm), t.okm);
         }
     }
+
+    #[test]
+    fn test_derive_sha1_with_none() {
+        let ikm = hex::decode("0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c").unwrap();
+        let salt = None;
+        let info = hex::decode("").unwrap();
```

Comment:
@warner `info: "",` can also be not set, does this have the same problems as the `salt`?